ocs_ci/helpers/disruption_helpers.py

```diff
@@ -24,6 +25,18 @@ class Disruptions:
     daemon_pid = None
 
     def set_resource(self, resource, leader_type="provisioner"):
+        # Managed services multicluster run - Use provider cluster
+        # TODO: Create a better solution to switch context when needed
+        context_switched = False
+        if (
+            config.multicluster
+            and get_primary_cluster_config().ENV_DATA.get("cluster_type") == "consumer"
+        ):
+            log.debug("Switching to provider")
+            config.switch_ctx(
+                get_provider_cluster_config().MULTICLUSTER["multicluster_index"]
+            )
+            context_switched = True
```

Comment:
this code is repeated multiple time, please create a function for it and call from all relevant places

Reply:
Working on updating the code. This code segment may change.